erpnext/setup/utils.py

```diff
@@ -107,6 +107,8 @@ def get_exchange_rate(from_currency, to_currency, transaction_date=None, args=No
 			# expire in 6 hours
 			response.raise_for_status()
 			value = response.json()["result"]
+			if not value and frappe.flags.in_test:
+				return 1.00
```

Comment:
Ideally, I would use a library like https://pypi.org/project/responses/ for mocking HTTP response from API, but adding another dependency just for one test doesn't feel justified. 

In tests we need to validate two things:
- Code is creating appropriate requests to the API. 
- Code is acting properly on valid/invalid responses given by API.

You could patch out the `request` library's `get` call and validate both requirements mentioned above. Essentially creating a fake API that validates input and gives output.